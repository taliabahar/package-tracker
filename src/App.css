body {
  height: 100vh;
  width: 100vw;
  background: #b2e7f3;
  height: 100vh;
  margin: 0;
}

.app {
  padding: 30px;
}

#header {
  text-align: center;
  color: white;
  font-size: 60px;
  text-shadow: 4px 5px #d9d9d9;
  font-family: "Shrikhand", cursive;
  margin: 0;
}

.todo-list {
  margin: 0 auto;
  /* background: #b2e7f3; */
  /* background: yellow; */
  border-radius: 4px;
  padding: 5px;
  max-width: 850px;
  min-height: 100%;
  margin-bottom: -320px;
  opacity: 1;
}

.packageInput {
  background: #fff;
  box-shadow: 1px 1px 1px rgba(0, 0, 0, 0.15);
  padding: 20px 30px;
  font-size: 15px;
  margin-bottom: 30px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  border: solid 3px #d9d9d9;
  border-radius: 5px;
  font-family: "Poppins", sans-serif;
}

.carrierOptions {
  margin-right: 40px;
  width: 120px;
  height: 30px;
}

input[type="text"] {
  margin-right: 30px;
  width: 200px;
  height: 25px;
}

#removeImg {
  cursor: pointer;
  width: 35px;
  height: 35px;
  margin-top: 5px;
}

.removeButton {
  float: left;
  width: 45px;
  height: 45px;
  border-radius: 100%;
  cursor: pointer;
  border: none;
}

#buttonText {
  font-size: 6px;
  color: black;
  font-weight: bold;
  text-align: center;
}

#boxImage {
  cursor: pointer;
}

.addPackageButton {
  position: relative;
  top: 110px;
  background-color: #b2e7f3;
  border: none;
  cursor: pointer;
}

.input {
  margin-top: 20px;
}

.track {
  border-radius: 10px;
  text-align: center;
  width: 60px;
  height: 40px;
  margin-left: -25px;
  cursor: pointer;
  background-color: white;
  font-family: "Roboto Condensed";
  padding: 3px;
  color: #000;
  box-shadow: 0px 8px 15px rgba(0, 0, 0, 0.1);
  transition: all 0.3s ease 0s;
  cursor: pointer;
  outline: none;
  text-transform: uppercase;
}

.track:enabled:hover {
  background-color: rgba(0, 255, 0, 0.5);
  box-shadow: -8px 2px 18px rgba(46, 229, 157, 0.4);
  transform: translateX(8px);
  font-weight: bold;
}

button[disabled],
button[disabled]:hover {
  border: 1px solid #999999;
  background-color: #cccccc;
  color: #666666;
}

#footerGraphic {
  overflow: scroll;
  position: fixed;
  z-index: -1;
  width: 100vw;
  height: 320px;
  bottom: 0;
  left: 0;
}

.footer .push {
  height: 50px;
}

.footer {
  line-height: 50px;
}
