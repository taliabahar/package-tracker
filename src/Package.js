import React, {useState, useEffect} from "react";
import removeIcon from './images/removeIcon.png'
import trackIcon from './images/tracking.png'

import r1 from './images/delete.png'
import r2 from './images/cross-remove-sign.png'
import r3 from './images/x-button.png'


import Modal from "./Modal";
import useModal from './useModal';
import {trackPackage} from './shippoAPI'

export function Package(props) {
    const {carrier, trackingNum, trackingName} = {props};
    const {isShowing, toggle} = useModal();
    
    const [status, setStatus] = useState('');
    const [statusDetails, setStatusDetails] = useState('');
    const [statusIndicatorColor, setStatusIndicatorColor] = useState('');

    // const isEnabled = trackingName.length > 0 && trackingNum.length > 0 && carrier.length > 0;
    const isEnabled = trackingName !== "" && trackingNum !== "" && carrier !== "";

    async function grabData() {
        const data = await trackPackage();
        const currentTrackingHistoryItem = data.data[0].tracking_history[data.data[0].tracking_history.length-1]
        setStatus(currentTrackingHistoryItem.status);
        setStatusDetails(currentTrackingHistoryItem.status_details);
    }

    useEffect(() => {
        statusIndicator();
    });

    // async function grabPackageData() {
    //     const data = await trackPackage();
    //     const currentTrackingLocation = data.data[0].tracking_history[data.data[0].tracking_history.length-1]
    //     setPackageCity(currentTrackingLocation.location.city);
    //     setPackageCountry(currentTrackingLocation.location.country);
    //   }
      
    //   async function grabLocationData() {
    //     const data = await getPackageCoordinates();
    //     setCenter(data.features[0].geometry.coordinates);
    //     console.log('HELLO', center)    
    //   }

    function statusIndicator() {
        if(status === "DELIVERED") {
            setStatusIndicatorColor("positive");
        } else if(status === "TRANSIT") {
            setStatusIndicatorColor("active");
        } else if(status === "FAILURE") {
            setStatusIndicatorColor("negative");
        } 
    }
    
    // onChange={(e) => setTrackingNumber(e.target.value) } 
    // onChange={(e) => setTrackingName(e.target.value) }
    // onChange={(e) => setCarrier(e.target.value) } 

    return (
        <div className="packageInput">
            <div>
            <form>
                <select className="carrierOptions" value={props.pkg.carrier} onChange={(e) => props.setCarrier(props.index, e.target.value) }>
                    <option value="" disabled selected>Select a Carrier</option>
                    <option value="usps">USPS</option> 
                    <option value="usp">UPS</option> 
                    <option value="fedex">FedEx</option> 
                </select>
                <input value={props.pkg.trackingNum} onChange={(e) => props.setTrackingNumber(props.index, e.target.value) } type="text" placeholder="Enter Package Tracking Number"></input>
                <input value={props.pkg.trackingName} onChange={(e) => props.setTrackingName(props.index, e.target.value) } type="text" placeholder="Enter Package Name"></input>
            </form>
            </div>
            <Modal isShowing={isShowing} hide={toggle} name={trackingName}
              carrier={carrier} trackingNum={trackingNum}
              status={status} statusDetails={statusDetails} statusIndicatorColor={statusIndicatorColor}
            />
            <button disabled={!isEnabled} onClick={() => {grabData(); toggle();}} className="track"> Track <img src={trackIcon} width="20px" height="20px" alt="tracking icon"></img></button>     
            <button className="removeButton" onClick={() => props.deletePackage(props.index)}><img id="removeImg" src={r3} alt="Remove Icon"></img></button>  
        </div>   
    )
}
