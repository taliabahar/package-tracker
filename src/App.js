import React, { useState, useEffect } from 'react';
import './App.css';
import {Package} from "./PackageInput";
import { apiHost } from './shippoAPI';
import box from './images/box2.png';

function App() {
  useEffect(() => apiHost('https://api.goshippo.com/tracks/'))

  const [packages, setPackages] = useState([
    {
      carrier: "",
      trackingNum: "",
      trackingName: ""
    }
  ])
  const addPackage = (pkg) => {
    const newPackage = [pkg, ...packages];
    setPackages(newPackage);
  }

  // filter by tracking number 

  // const deletePackage= (e) => {
  //   const name = e.target.getAttribute("trackingNumber")
  //    setPackages(packages.filter(item => item.name !== name));
  //  };
 
 
  const deletePackage = index => {
    const newPackages = [...packages];
    newPackages.splice(index, 1);
    setPackages(newPackages);
    // setPackages(newPackages.filter(packageBox => packageBox.index !== index))
  }

  return (
    <div className = "app">
      <link href="https://fonts.googleapis.com/css?family=Barlow+Condensed|Lexend+Deca|Modak|News+Cycle|Nunito+Sans|Poppins|Quattrocento+Sans|Roboto+Condensed|Shrikhand&display=swap" rel="stylesheet"></link>
          <h1 id="header">Package Tracker</h1>
      <div className="add">
      <button className="addPackageButton" onClick={(e) => addPackage(e)}><img id="boxImage" src={box} alt="box" width="100px" height="100px"></img></button>
      </div>
      <div className = "todo-list">
        {packages.map((packages, index) => (
          console.log("INDEX" + index),
          <Package index={index} deletePackage={deletePackage} /> 
        ))}
      </div>
    </div>
  );
}

export default App;
