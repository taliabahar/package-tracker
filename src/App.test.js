import React, { useState } from "react";
import ReactDOM from "react-dom";
import App from "./App";
import TestRenderer from "react-test-renderer";
// import { App } from "./App";
import { Package } from "./Package";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("the app should initialize with one empty package form", () => {
  const component = TestRenderer.create(<App />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("the app should initialize with the add package button", () => {
  const component = TestRenderer.create(<App />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

//TESTS:
// remove removes right one
// add packages button adds package
