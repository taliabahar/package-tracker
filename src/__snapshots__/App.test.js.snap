// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the app should initialize with one empty package form 1`] = `
<div
  className="app"
>
  <link
    href="https://fonts.googleapis.com/css?family=Roboto+Condensed|Shrikhand&display=swap"
    rel="stylesheet"
  />
  <h1
    id="header"
  >
    Package Tracker
  </h1>
  <div
    className="add"
  >
    <button
      className="addPackageButton"
      onClick={[Function]}
    >
      <img
        alt="box"
        height="100px"
        id="boxImage"
        src="boxGreen.png"
        width="100px"
      />
    </button>
  </div>
  <div
    className="todo-list"
  >
    <div
      className="packageInput"
    >
      <div>
        <form>
          <select
            className="carrierOptions"
            id="carrierInput"
            onChange={[Function]}
            value=""
          >
            <option
              disabled={true}
              selected={true}
              value=""
            >
              Select a Carrier
            </option>
            <option
              value="usps"
            >
              USPS
            </option>
            <option
              value="usp"
            >
              UPS
            </option>
            <option
              value="fedex"
            >
              FedEx
            </option>
          </select>
          <input
            id="trackingNumberInput"
            onChange={[Function]}
            placeholder="Enter Package Tracking Number"
            type="text"
            value=""
          />
          <input
            id="trackingNameInput"
            onChange={[Function]}
            placeholder="Enter Package Name"
            type="text"
            value=""
          />
        </form>
      </div>
      <button
        className="track"
        disabled={true}
        onClick={[Function]}
      >
         
        Track
         
        <img
          alt="tracking icon"
          height="20px"
          src="tracking.png"
          width="20px"
        />
      </button>
      <button
        className="removeButton"
        onClick={[Function]}
      >
        <img
          alt="Remove Icon"
          id="removeImg"
          src="x-button.png"
        />
      </button>
    </div>
    <div
      className="push"
    />
  </div>
  <footer
    className="footer"
  >
    <img
      alt="Truck and hills"
      id="footerGraphic"
      src="footer.png"
    />
  </footer>
</div>
`;

exports[`the app should initialize with the add package button 1`] = `
<div
  className="app"
>
  <link
    href="https://fonts.googleapis.com/css?family=Roboto+Condensed|Shrikhand&display=swap"
    rel="stylesheet"
  />
  <h1
    id="header"
  >
    Package Tracker
  </h1>
  <div
    className="add"
  >
    <button
      className="addPackageButton"
      onClick={[Function]}
    >
      <img
        alt="box"
        height="100px"
        id="boxImage"
        src="boxGreen.png"
        width="100px"
      />
    </button>
  </div>
  <div
    className="todo-list"
  >
    <div
      className="packageInput"
    >
      <div>
        <form>
          <select
            className="carrierOptions"
            id="carrierInput"
            onChange={[Function]}
            value=""
          >
            <option
              disabled={true}
              selected={true}
              value=""
            >
              Select a Carrier
            </option>
            <option
              value="usps"
            >
              USPS
            </option>
            <option
              value="usp"
            >
              UPS
            </option>
            <option
              value="fedex"
            >
              FedEx
            </option>
          </select>
          <input
            id="trackingNumberInput"
            onChange={[Function]}
            placeholder="Enter Package Tracking Number"
            type="text"
            value=""
          />
          <input
            id="trackingNameInput"
            onChange={[Function]}
            placeholder="Enter Package Name"
            type="text"
            value=""
          />
        </form>
      </div>
      <button
        className="track"
        disabled={true}
        onClick={[Function]}
      >
         
        Track
         
        <img
          alt="tracking icon"
          height="20px"
          src="tracking.png"
          width="20px"
        />
      </button>
      <button
        className="removeButton"
        onClick={[Function]}
      >
        <img
          alt="Remove Icon"
          id="removeImg"
          src="x-button.png"
        />
      </button>
    </div>
    <div
      className="push"
    />
  </div>
  <footer
    className="footer"
  >
    <img
      alt="Truck and hills"
      id="footerGraphic"
      src="footer.png"
    />
  </footer>
</div>
`;
